src/volatile_memory.rs

```diff
@@ -2044,16 +2043,15 @@ mod tests {
         } else {
             File::open(Path::new("c:\\Windows\\system32\\ntoskrnl.exe")).unwrap()
         };
-        assert!(s.read_exact_from(2, &mut file, size_of::<u32>()).is_err());
-        assert!(s
-            .read_exact_from(core::usize::MAX, &mut file, size_of::<u32>())
-            .is_err());
```

Comment:
This test case was dropped, right? Is it no longer needed?

Reply:
Yes, these test cases were checking the out-of-bounds checks in read_exact_from, however using `ReadVolatile::read_exact_volatile` does not take offset/count parameters and instead expects the `VolatileSlice` being passed to have the correct offset/size. This moves the bounds checks into the `.get_slice()` call, which is a separate function with separate tests (e.g. its impossible to replicate this test scenario with the new APIs)